spec.md

```diff
@@ -288,6 +289,31 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+# Size Constraints
+
+In many scenarios, CloudEvents will be forwarded through generic middleware,
+and potentially even via multiple routing hops. CloudEvents might also be
+routed to consumers, like embedded devices, that are very memory-constrained.
+
+To ensure interoperability for such scenarios, CloudEvents publishers SHOULD
+avoid embedding large data items into event payloads and rather use the event
+payload to point to such data items. This also allows for a broader distribution
+of events, because the referenced external data items allow for differentiated
+access control to sensitive details.
+
+The overall wire-size of a published event SHOULD NOT exceed 64 KByte. This
```

Comment:
Can we re-word this more along the lines of _"... CloudEvent intermediaries MUST be capable of transporting and delivering CloudEvents with a wire-size of 64 KByte.." - we're not looking to limit the size and i believe SHOULD NOT is a bit of a strong statement.